source/frontend/src/features/documents/documentDetail/DocumentDetailForm.tsx

```diff
@@ -58,6 +60,36 @@ export const DocumentDetailForm: React.FunctionComponent<
     [props.documentTypes],
   );
 
+  const [documentTypePurpose, setDocumentTypePurpose] = useState(null);
+
+  const matchDocumentType = useCallback(
+    (documentTypeId: number) => {
+      const purpose = props.documentTypes.find(
+        x => x.id === Number(documentTypeId),
+      )?.documentTypePurpose;
+
+      setDocumentTypePurpose(purpose);
+    },
+    [props.documentTypes],
+  );
+
+  useEffect(() => {
```

Comment:
hmm, I'm not sure why this one is here, since this is a ref and won't trigger useEffect updates. Why is this necessary when the event handler on line 81 exists? Is this just a useEffect that is intended to fire once during init? If so, I'm not sure if a useEffect is necessary.

Reply:
Yes, its for the first render, otherwise the purpose is not set from the other function